src/logic/hydra.lean

```diff
@@ -174,8 +174,7 @@ begin
   induction hacc with a h ih,
   refine acc.intro _ (λ s, _),
   classical, rw cut_expand_iff hi,
-  rintro ⟨t, a, hr, ha, rfl⟩,
-  cases mem_singleton.1 ha,
+  rintro ⟨t, a, hr, (rfl|⟨⟨⟩⟩), rfl⟩,
```

Comment:
IMHO, `(rfl : a = _)` looks better than `(rfl|⟨⟨⟩⟩)`. Otherwise LGTM. Thanks!
bors d+

Reply:
You mean replace the `rfl` by `rfl : a = _`? Seems verbose. `x ∈ {a}` is defeq to `x = a ∨ x ∈ ∅` so _|_ is needed. (The parentheses are actually not needed.)